core/tx_pool.go

```diff
@@ -963,6 +965,44 @@ func (pool *TxPool) promoteTx(addr common.Address, hash common.Hash, tx *types.T
 	return true
 }
 
+func (pool *TxPool) OrderBookTxs() map[common.Address]types.Transactions {
+	txs := map[common.Address]types.Transactions{}
+	for from, txList := range pool.orderBookQueue {
+		txs[from] = txList.Flatten()
+	}
+	return txs
+}
+
+func (pool *TxPool) PurgeOrderBookTxs() {
+	for from, _ := range pool.orderBookQueue {
+		delete(pool.orderBookQueue, from)
+	}
+}
+
+func (pool *TxPool) GetOrderBookTxNonce(address common.Address) uint64 {
+	nonce := pool.Nonce(address)
+	val, ok := pool.orderBookQueue[address]
+	if ok {
+		return nonce + uint64(val.Len())
+	}
+	return nonce
+}
+
+func (pool *TxPool) AddOrderBookTx(tx *types.Transaction) error {
+	if from, err := types.Sender(pool.signer, tx); err == nil {
+		val, ok := pool.orderBookQueue[from]
+		if !ok {
+			val = newTxList(false)
+			pool.orderBookQueue[from] = val
+		}
+		ok, _ = val.Add(tx, 0)
```

Comment:
does this function return error too?

Reply:
nope. It just returns false if unable to add tx to list.